src/data_collection.py

```diff
@@ -228,106 +268,170 @@ def calculate_individual_metrics(json_file_name="raw_data_storage"):
+    # ask the user for a URL of a Github repository
+    miner = RepositoryMining(path_to_repo=repo)
+    return miner
 
-# Steps to run data_collection.py and display the data table:
 
-# Must be in the repository folder
-# Run the python file using `pipenv run python src/data_collection.py`
-# Enter the name of the json file you want the data to be written to
-# If prompted, due to data not being collected previously, enter the URL
-# or path of the Repository you would like to analyze
-# If the current repository is the one you would like to analyze, simply
-# hit enter wihtout typing anything.
+def merge_metric_and_issue_dicts(metrics_dict, issues_dict):
+    """Receive two dicitionaries one for issues and the other for metrics.
 
+    Create empty fields for users existing in issues_dict and not in metrics.
+    """
+    for entry in issues_dict:
+        # check if the issue/PR author does not exist in the metrics dicitionary
+        if entry not in metrics_dict.keys():
+            # Add empty data to their metrics
+            metrics_dict[entry] = {
+                "EMAIL": "N/A",
+                "COMMITS": 0,
+                "ADDED": 0,
+                "REMOVED": 0,
+                "TOTAL": 0,
+                "MODIFIED": 0,
+                "RATIO": 0,
+                "FILES": [],
+                "FORMAT": [],
+            }
+        # update the metrics dicitionary with the new keys
+        metrics_dict[entry].update(issues_dict[entry])
+    return metrics_dict
+
+
+def merge_duplicate_usernames(dictionary, kept_entry, removed_entry):
+    """Take input from user and merge data in entries then delete one."""
+    # Put the keys of mergable metrics in a list
+    categories = [
+        "COMMITS",
+        "ADDED",
+        "REMOVED",
+        "FILES",
+        "issues_commented",
+        "issues_opened",
+        "pull_requests_opened",
+        "pull_requests_commented",
+    ]
+    # Loop through all the keys in the list
+    for category in categories:
+        # Special case for merging files to avoid duplicates
+        if category == "FILES":
+            dictionary[kept_entry][category] = list(
+                set(dictionary[kept_entry][category])
+                | set(dictionary[removed_entry][category])
+            )
+            # sort the files for testing consistency
+            dictionary[kept_entry][category] = sorted(dictionary[kept_entry][category])
+        else:
+            # simple addition for all other metrics
+            dictionary[kept_entry][category] += dictionary[removed_entry][category]
+    # Use exception handling to delete the duplicate from the dictionary
+    try:
+        del dictionary[removed_entry]
+    except KeyError:
+        print("Key 'testing' not found")
+    return dictionary
+
+
+# This main method is only for the purposes of testing
+# Main methods should only be in the cogitate.py file
 if __name__ == "__main__":
```

Comment:
Issue tracker #78 brought up that the `cogitate.py` file should be one of the only files with a main method. Could this code be moved to a function?

Reply:
The main function will be removed once the CLI team merge their changes, otherwise, there will not be a way to run the tool